src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -160,37 +179,33 @@ internal void SetSearchPathBuildProperty(string path)
 
     public void AddBuildPhaseSymbolUpload(SentryCliOptions sentryCliOptions)
     {
-        _logger?.LogInfo("Adding automated debug symbol upload script to build phase.");
+        _logger?.LogInfo("Setting up Sentry CLI build properties and upload script.");
 
-        if (MainTargetContainsSymbolUploadBuildPhase())
-        {
-            _logger?.LogDebug("Build phase '{0}' already added.", SymbolUploadPhaseName);
-            return;
-        }
+        _pbxProjectType.GetMethod("SetBuildProperty", new[] { typeof(string), typeof(string), typeof(string) })
+            .Invoke(_project, new object[] { _mainTargetGuid, SymbolUploadPropertyName, sentryCliOptions.UploadSymbols ? "YES" : "NO" });
 
-        var uploadDifArguments = "--il2cpp-mapping";
-        if (sentryCliOptions.UploadSources)
-        {
-            uploadDifArguments += " --include-sources";
-        }
+        // Set additional upload properties
+        _pbxProjectType.GetMethod("SetBuildProperty", new[] { typeof(string), typeof(string), typeof(string) })
+            .Invoke(_project, new object[] { _mainTargetGuid, IncludeSourcesPropertyName, sentryCliOptions.UploadSources ? "YES" : "NO" });
+
+        _pbxProjectType.GetMethod("SetBuildProperty", new[] { typeof(string), typeof(string), typeof(string) })
+            .Invoke(_project, new object[] { _mainTargetGuid, AllowFailurePropertyName, sentryCliOptions.IgnoreCliErrors ? "YES" : "NO" });
 
-        if (sentryCliOptions.IgnoreCliErrors)
+        _pbxProjectType.GetMethod("SetBuildProperty", new[] { typeof(string), typeof(string), typeof(string) })
+            .Invoke(_project, new object[] { _mainTargetGuid, PrintLogsPropertyName, sentryCliOptions.IgnoreCliErrors ? "NO" : "YES" });
+
+        if (MainTargetContainsSymbolUploadBuildPhase())
         {
-            uploadDifArguments += " --allow-failure";
+            _logger?.LogInfo("Success. Build phase '{0}' was already added.", SymbolUploadPhaseName);
+            return;
         }
 
-        var uploadScript = string.Format(_uploadScript,
-            SentryCli.SentryCliMacOS,
-            uploadDifArguments,
-            // Xcode parses the log-messages for 'error:', causing the phase to error with
-            // 'Command PhaseScriptExecution emitted errors but did not return a nonzero exit code to indicate failure'
-            // even with the '--allow-failure' arg. In that case, we're just logging to file instead.
-            sentryCliOptions.IgnoreCliErrors ? LogOnlyArg : LogAndPrintArg);
+        _logger?.LogDebug("Adding the upload script to {0}.", SymbolUploadPhaseName);
```

Comment:
These properties can get overwritten via Unity's PBXProject. The shellscript itself cannot. Attempting to overwrite the script causes unexpected behaviour like adding it twice.

Instead, the properties get set (and overwritten) during the Unity export and read by the script during the Xcode build